docs/nydus-dev.md

```diff
@@ -0,0 +1,60 @@
+## Architecture Overview
```

Comment:
Great job! Maybe we can reference it from README.md.

Reply:
Let's add more content to the nydus-dev doc, the add reference to it from README.md?